sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFileSuite.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.execution
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.scalatest.GivenWhenThen
+import org.scalatestplus.mockito.MockitoSugar
+
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.catalyst.catalog.CatalogTable
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.hive.test.TestHiveSingleton
+
+class SaveAsHiveFileSuite extends QueryTest with TestHiveSingleton
+    with MockitoSugar with GivenWhenThen {
+  test("getMRTmpPath method") {
+    val insertIntoHiveTable = new InsertIntoHiveTable(
+      mock[CatalogTable], Map.empty[String, Option[String]],
+      mock[LogicalPlan], true, false, Seq.empty[String])
+    val hadoopConf = new Configuration()
+    val scratchDir = "/tmp/hive_scratch"
+    val sessionScratchDir = "/tmp/hive/user_a/session_b"
+
+    Given(s"sessionScratchDir = '$sessionScratchDir' & scratchDir = '$scratchDir'")
+    When("get the path from getMRTmpPath(hadoopConf, sessionScratchDir, scratchDir)")
```

Comment:
Can you follow the format of the other tests? For example, if you have no strong opinion, I think you'd be better to just use `assert` instead of `GivenWhenThen` to follow the others.

Reply:
I see, I changed to use simple `assert` instead of `GivenWhenThen`.